File: src/main/java/controller/GamesController.java
package controller;

import model.Game;
import service.JsonService;
import service.SteamWishlistService;
import spark.Request;
import spark.Response;
import spark.Route;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

public class GamesController {
    private static List<Game> gameList = new ArrayList<>();
    private static LocalDateTime gameListUpdate;

    private static synchronized List<Game> getGameList() {
        if (gameListUpdate == null || gameListUpdate.isBefore(LocalDateTime.now().minusHours(1))) {
            gameList = SteamWishlistService.getWishlist("loll3k");
            gameListUpdate = LocalDateTime.now();
        }
        return gameList;
    }

    public static Route index = (Request request, Response response) -> JsonService.toJson(getGameList());
}

File: src/main/kotlin/controller/GamesController.kt
package controller

import model.Game
import service.JsonService
import service.SteamWishlistService
import spark.Request
import spark.Response
import spark.Spark

import java.time.LocalDateTime

class GamesController {
    companion object {

        fun register(endpointUrl: String) {
            Spark.get(endpointUrl, index, JsonService.gson::toJson)
        }

        private var gameListUpdate: LocalDateTime? = null
        private var gameList: List<Game> = listOf()

        val index = fun(_: Request, _: Response): List<Game> {
            synchronized(this) {
                if (gameListUpdate?.isBefore(LocalDateTime.now().minusHours(1)) ?: true) {
                    gameList = SteamWishlistService.getWishlist("loll3k")
                    gameListUpdate = LocalDateTime.now()
                }
            }
            return gameList
        }
    }
}
